api/src/main/java/javax/enterprise/inject/spi/builder/BeanAttributesConfigurator.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * @see ProcessBeanAttributes#configureBeanAttributes()
+ * @param <T> the class of the bean instance
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public interface BeanAttributesConfigurator<T> {
+    /**
+     * Read the information from the given {@link AnnotatedType}. All relevant information is overwritten.
+     *
+     * @param type AnnotatedType to read information from
+     * @return self
+     */
+    <U extends T> BeanAttributesConfigurator<U> read(AnnotatedType<U> type);
+
+    /**
+     * Read the information from the given {@link AnnotatedMember} (field or method).
+     * All relevant information is overwritten.
+     *
+     * @param member AnnotatedMember to read information from
+     * @return self
+     */
+    <U extends T> BeanAttributesConfigurator<U> read(AnnotatedMember<U> member);
+
+    /**
+     * Read the information from the given bean attributes. All relevant information is overwritten.
+     *
+     * @param beanAttributes beanAttributes to read information from
+     * @return self
+     */
+    BeanAttributesConfigurator<T> read(BeanAttributes<?> beanAttributes);
+
+    /**
+     *
+     * Add a type to the bean types
+     *
+     * @param type the type to add
+     * @return self
+     */
+    BeanAttributesConfigurator<T> addType(Type type);
+
+    /**
+     *
+     * Add a type to the bean types
+     *
+     * @param typeLiteral the type to add
+     * @return self
+     */
+    BeanAttributesConfigurator<T> addType(TypeLiteral<?> typeLiteral);
+
+    /**
+     *
+     * Add types to the bean types
+     *
+     * @param types types to add
+     * @return self
+     */
+    BeanAttributesConfigurator<T> addTypes(Type... types);
+
+    /**
+     *
+     * Add types to the bean types
+     *
+     * @param types types to add
+     * @return self
+     */
+    BeanAttributesConfigurator<T> addTypes(Set<Type> types);
+
+    /**
+     * Adds an unrestricted set of bean types for the given type as if it represented a bean class of a managed bean.
+     * Illegal bean types are omitted.
+     *
+     * @param type to build the closure from
+     * @return self
+     */
+    BeanAttributesConfigurator<T> addTransitiveTypeClosure(Type type);
+
+    /**
+     *
+     * Replace bean types
+     *
+     * @param types the types of the configured bean
+     * @return self
+     */
+    BeanAttributesConfigurator<T> types(Type... types);
+
+    /**
+     *
+     * Replace bean types
+     *
+     * @param types the types of the configured bean
+     * @return self
+     */
+    BeanAttributesConfigurator<T> types(Set<Type> types);
+
+    /**
+     *
+     * Replace Bean scope
+     *
+     * @param scope new scope for the configured bean
+     * @return self
+     */
+    BeanAttributesConfigurator<T> scope(Class<? extends Annotation> scope);
+
+    /**
+     *
+     * Add a qualifier to the configured bean
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
```

Comment:
It's not a builder. We should state (explicitly in the spec) that `@Default` is not removed when you add `@Named`.